nilearn/datasets/tests/test_func.py

```diff
@@ -0,0 +1,345 @@
+"""
+Test the datasets module
+"""
+# Author: Alexandre Abraham
+# License: simplified BSD
+
+import os
+import shutil
+import numpy as np
+from tempfile import mkdtemp
+
+from nose import with_setup
+from nose.tools import assert_true, assert_equal, assert_raises
+
+
+from nilearn import datasets
+from nilearn._utils.testing import (mock_request, wrap_chunk_read_,
+                                    FetchFilesMock)
+
+from nilearn._utils.compat import _basestring
+
+
+currdir = os.path.dirname(os.path.abspath(__file__))
+datadir = os.path.join(currdir, 'data')
+tmpdir = None
+url_request = None
+file_mock = None
+
+
+def setup_tmpdata():
+    # create temporary dir
+    global tmpdir
+    tmpdir = mkdtemp()
+
+
+def setup_mock():
+    global url_request
+    url_request = mock_request()
+    datasets.utils._urllib.request = url_request
+    datasets.utils._chunk_read_ = wrap_chunk_read_(datasets.utils._chunk_read_)
+    global file_mock
+    file_mock = FetchFilesMock()
+    datasets.func._fetch_files = file_mock  # overwrite the actual function
+
+
+def teardown_tmpdata():
+    # remove temporary dir
+    global tmpdir
+    if tmpdir is not None:
+        shutil.rmtree(tmpdir)
+
+
+@with_setup(setup_mock)
```

Comment:
The point of this test is to run the non-mocked code. This annotation makes the test pointless and reduce code coverage.